.github/commit-and-push-gherkin-update.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+
+if [ "$PR_EXISTS" == 'false' ]; then
+    echo "Creating branch $BRANCH_NAME"
+    git checkout -b "$BRANCH_NAME"
+fi
+
+echo "Creating commit"
+git add -u
+git -c user.name='github-actions[bot]' \
+    -c user.email='41898282+github-actions[bot]@users.noreply.github.com' \
+    commit \
+    -m "$COMMIT_MSG"
+
+echo "Authenticating and pushing"
+# This will reconfigure git to push with the token provided in the GH_TOKEN env var
+gh auth setup-git
+git push -u origin "$BRANCH_NAME"
```

Comment:
Design decision: the update branches will be in the main `Behat/Gherkin` repository (as they were before). We'll need to relax the branch protection rules to allow us to delete `cucumber-update-*` branches after they're merged. This does mean they'll appear in packagist.

Alternatively we could create a `Behat/Gherkin-Automation-PRS` repo as a fork and configure our github app with the permissions to push the branch there instead. However that complicates things a bit and I think it's simpler to keep everything together?